src/Schema/SQLServerSchemaManager.php

```diff
@@ -85,12 +85,12 @@ public function listTableForeignKeys($table, $database = null)
      */
     public function listSchemaNames(): array
     {
+        // Exclude internal (<5) and backward compatibility existed schemas (>16383)
         return $this->_conn->fetchFirstColumn(
             <<<'SQL'
 SELECT name
 FROM   sys.schemas
-WHERE  name NOT IN('guest', 'INFORMATION_SCHEMA', 'sys')
-AND    name NOT IN(SELECT name FROM sys.database_principals WHERE type='R')
+WHERE  schema_id BETWEEN 5 AND 16383
```

Comment:
If it works for you, please use the query from https://github.com/doctrine/dbal/pull/5739#discussion_r991822084.

We should refrain from using magic constants unless using them is recommended in the official documentation.